addons/binding/org.openhab.binding.autelis/src/main/java/org/openhab/binding/autelis/handler/AutelisHandler.java

```diff
@@ -0,0 +1,401 @@
+
+	public AutelisHandler(Thing thing) {
+		super(thing);
+	}
+
+	@Override
+	public void initialize() {
+		super.initialize();
+		configure();
+	}
+
+	@Override
+	public void dispose() {
+		logger.debug("Handler disposed.");
+		stopPolling();
+	}
+
+	@Override
+	public void handleCommand(ChannelUID channelUID, Command command) {
+		
+		if(channelUID.getId().equals("lightscmd")){
+			/*
+			 * lighting command
+			 * possible values, but we will let anything through.
+			 * alloff, allon, csync, cset, cswim, party, romance, caribbean, american,
+			 * sunset, royalty, blue, green, red, white, magenta, hold, recall
+			 */
+			getUrl( baseURL + "lights.cgi?val=" +  command.toString(), TIMEOUT);
+		} else {
+			String [] args = channelUID.getId().split("-");
+			if(args.length < 2){
+				logger.warn("Unown channel {} for command {}", channelUID, command);
+				return;
+			}
+			String type = args[0];
+			String name  = args[1];
+			
+			if(type.equals("equipment")){				
+				String cmd = "value";
+				int value;
+				if(command == OnOffType.OFF){
+					value = 0;
+				} else if(command == OnOffType.ON){
+					value = 1;
+				} else if(command instanceof DecimalType){
+					value = ((DecimalType)command).intValue();
+					if(value >= 3){
+						//this is a dim type. not sure what 2 does
+						cmd = "dim";
+					}
+				} else {
+					logger.error("command type {} is not supported", command);
+					return;
+				}
+				String response = getUrl(baseURL + "/set.cgi?name=" + name + "&" + cmd + "=" + value, TIMEOUT);
+				logger.trace("equipment set {} {} {} : result {}", name, cmd, value, response);
+			} else if(type.equals("temp")){
+				String value;
+				if(command == IncreaseDecreaseType.INCREASE){
+					value = "up";
+				}else if(command == IncreaseDecreaseType.DECREASE){
+					value = "down";
+				}else {
+					value = command.toString();
+				}
+				
+				String cmd;
+				//name ending in sp are setpoints, ht are heat types?
+				if(name.endsWith("sp")){
+					cmd = "temp";
+				} else if(name.endsWith("ht")){
+					cmd = "hval";
+				} else {
+					logger.error("Unknown temp type {}", name);
+					return;
+				}
+				
+				String response = getUrl(baseURL + "/set.cgi?wait=1&name=" + name + "&" + cmd + "=" + value, TIMEOUT);
+				logger.trace("temp set {} {} : result {}", cmd, value, response);
+			} else {
+				logger.error("Unsupported type {}", type);
+			}
+		}
+		scheduleClearTime(UPDATE_CLEARTIME);
+	}
+
+	/**
+	 * Configures this thing
+	 */
+	private void configure() {
+
+		properlyConfigured = false;
+
+		AutelisConfiguration configuration = getConfig().as(AutelisConfiguration.class);
+
+		port = configuration.port.intValue();
+		host = configuration.host;
+		username = configuration.user;
+		password = configuration.pass;
+		refresh = configuration.refresh.intValue();
+		
+		baseURL = "http://" + host + ":" + port;
+
+		logger.debug("Autelius binding configured with base url {}", baseURL);
+
+		properlyConfigured = true;
```

Comment:
You would actually need to check whether the parameters are correct, i.e. that you can access the controller and the credentials are valid. Depending on the result, you should call setThingStatus(ONLINE resp. OFFLINE).